nixos/modules/security/please.nix

```diff
@@ -0,0 +1,119 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.security.please;
+in
+{
+  options.security.please = {
+    enable = mkEnableOption (mdDoc ''
+      please, a tool which allows non-root users to execute commands as root
+    '');
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.please;
+      defaultText = literalExpression "pkgs.please";
+      description = mdDoc ''
+        Which package to use for {command}`please`;
+      '';
+    };
+
+    wheelAsRoot = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = lib.mdDoc ''
+          Whether users of the `wheel` group can execute commands or edit files
+          using {command}`please` and {command}`pleaseedit` respectively.
+        '';
+      };
+
+      needsPassword = mkOption {
+        type = types.bool;
+        default = true;
+        description = lib.mdDoc ''
+          Whether users of the `wheel` group must provide a password to run
+          commands or edit files {command}`please` and {command}`pleaseedit`
+          respectively.
+        '';
+      };
+    };
+
+    settings = mkOption {
+      type = types.attrs;
+      default = { };
+      example = {
+        jim_become_root = {
+          name = "jim";
+          target = "root";
+          rule = ".*";
+          require_pass = false;
+        };
+        jim_hosts = {
+          name = "jim";
+          type = "edit";
+          target = "root";
+          rule = "/etc/hosts";
+          editmode = 644;
+          require_pass = false;
+        };
+      };
+      description = mdDoc ''
+        Please configuration. Refer to
+        <https://github.com/edneville/please/blob/master/please.ini.md> for
+        details.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    security.wrappers =
+      let
+        owner = "root";
+        group = "root";
+        setuid = true;
+      in
+      {
+        please = {
+          source = "${cfg.package}/bin/please";
+          inherit owner group setuid;
+        };
+        pleaseedit = {
+          source = "${cfg.package}/bin/pleaseedit";
+          inherit owner group setuid;
+        };
+      };
+
+    environment = {
+      systemPackages = [ cfg.package ];
+
+      etc."please.ini" = {
+        text = generators.toINI { }
+          (
+            (if cfg.wheelAsRoot.enable then rec {
+              wheel_run_as_root = {
+                name = "wheel";
+                group = true;
+                target = "root";
+                rule = ".*";
+                require_pass = cfg.wheelAsRoot.needsPassword;
+              };
+              wheel_edit_as_root = wheel_run_as_root // { type = "edit"; };
+            } else { })
+            //
+            cfg.settings
+          );
```

Comment:
I can't make this work with `mkMerge` and `mkIf` for some unknown reason... It just keeps failing for me in `toINI` with:
```
error: value is a string while a set was expected

       at /nix/store/50vbgfbcrhlm2q3m111rn4nmxzw0svjv-source/lib/attrsets.nix:356:39:

          355|   mapAttrsToList = f: attrs:
          356|     map (name: f name attrs.${name}) (attrNames attrs);
             |                                       ^
          357|
(use '--show-trace' to show detailed location information)
```

Reply:
Take a look at my first patch (linked above) for an idea of how I'd go about doing this.